product/views.py

```diff
@@ -1,3 +1,43 @@
+import json
+
+from django.views     import View
+from django.http      import JsonResponse
 from django.shortcuts import render
+ 
+from .models     import Category, Subcategory, Product, ProductImage, Discount, Review
+from user.models import Member
+
+class ProductView(View):
+    def get(self, request, id):
+        try:
+            if Product.objects.filter(id=id) == "": #아무것도 반환하지 않을때 조건을 어떻게 명시해야하나
+                raise ValueError
+            product_list =  list(Product.objects.filter(id=id).values())
+            return JsonResponse({"message" : "SUCCESS", "result" : product_list}, status = 200) #리스트로 반환되는데 괜찮은건가
+        except ValueError:
+            return JsonResponse({'message' : 'ObjectDoesNotExist'}, status = 400)
+#한국어 인코딩이 안됨
+
+class CategoryView(View):
+    def get(self, request):
+            category_seq    = request.GET.get('category', None)
+            subcategory_seq = request.GET.get('subcategory', None)
+            sort            = request.GET.get('sort', None)
+
+            if subcategory_seq == None:
+                product_list =  list(Product.objects.filter(category=category_seq).order_by(sort).values())
+            else:
+                product_list =  list(Product.objects.filter(category=category_seq, subcategory=subcategory_seq).order_by(sort).values())
+
+            return JsonResponse({"message" : "SUCCESS", "result" : product_list}, status = 200)
+class ReviewView(View):
+    def post(self, request, id):
+        data        = json.loads(request.body)
+        
+        if Product.objects.filter(id=id) == "": #아무것도 반환하지 않을때 조건을 어떻게 명시해야하나
+            raise ValueError
+        product_ins = Product.objects.only('id').get(id=id)
+        member_ins  = Member.objects.only('id').get(id=data["member"])
+        Review.objects.create(star_rating=data['star_rating'], created_at=data['created_at'] , product=product_ins, member=member_ins, content = data['content']) 
```

Comment:
```suggestion
        if Product.objects.filter(id=product_id).exists():
            product_ins = Product.objects.only('id').get(id=id)
            member_ins  = Member.objects.only('id').get(id=data["member"])
            Review.objects.create(star_rating=data['star_rating'], created_at=data['created_at'] , product=product_ins, member=member_ins, content = data['content']) 
```
질문. 회원가입하지 않아도 제품에 리뷰를 달수 있나요? 아니라면 데코레이터를 통해서 유저를 받아오세요.
product_id의 검증이 끝났으면 굳이 Product 에서 다시 가져올 필요없이 Review create 시 product_id를 넘기면 될 것 같습니다.